File: firstFitAct.java
import java.util.*;

public class firstFitAct{

    public static Scanner sc = new Scanner(System.in);

    public static void main (String args[]) {
        int nBlock, nProcess, sizeBlock, sizeProcess, block[], process[], allocation[];
        String sBlock, sProcess;
        System.out.print("Enter the number of memory blocks: ");
        sBlock = sc.next();
        nBlock = Integer.parseInt(sBlock);
        System.out.println("--------------------");
    
        block = new int[nBlock];
        allocation = new int[nBlock];
        for (int i = 0; i < nBlock; i++) {
            System.out.print("Enter the size of memory block " + (i+1) + ": ");
            sizeBlock = sc.nextInt();
            block[i] = sizeBlock;
        }
        
        System.out.print("Enter the number of processes: ");
        sProcess = sc.next();
        nProcess = Integer.parseInt(sProcess);
        System.out.println("--------------------");
        
        process = new int[nProcess ];
        for (int i = 0; i < nProcess; i++) {
            System.out.print("Enter the size of file process " + (i+1) + ": ");
            sizeProcess = sc.nextInt();
            process[i] = sizeProcess;
        }

        System.out.println("--------------------INPUTS");
        for (int i = 0; i < nBlock; i++) {
            System.out.println("Block " + (i+1) + ": " + block[i]);
        }
        for (int i = 0; i < nProcess; i++) {
            System.out.println("Process " + (i+1) + ": " + process[i]);
        }

        for (int i = 0; i < nBlock; i++) {
            allocation[i] = -1;
        }

        for (int i = 0; i < nProcess; i++) {
            for (int j = 0; j < nBlock; j++) {
                if (block[j] >= process[i]) {
                    allocation[i] = j;
                    block[j] -= process[i];
                    break;
                }
            }
        }

        System.out.println("--------------------OUTPUTS");
        System.out.println("Process No.\tProcess Size\tBlock No.\tTransferred Process Size\tRemaining Block Size");
        for (int i = 0; i < nProcess; i++) {
            System.out.print("     " + (i+1) + "\t\t     " + process[i] + "\t   ");
            if (allocation[i] != -1) {
                System.out.println((allocation[i] + 1) + "\t\t\t" + process[i] + "\t\t\t" + block[i]);
            } else {
                System.out.println("Not Allocated" + "\t\t-" + "\t\t\t-");
            }
        }
    }
}

File: first-fit-act.c
#include<stdio.h>

int main()
{
    //initializing variables
    int nBlock, nProcess, sizeBlock, sizeProcess;

    //taking input from user
    printf("Enter the number of blocks in the memory: ");
    //using scanf to take input from user
    scanf("%i", &nBlock);
    //declaring array
    int blockSize[nBlock];
    int allocation[nBlock];
    //using for loop to take inputs
    printf("Enter the number of processes in the input queue: ");
    scanf("%i", &nProcess);
    int processSize[nProcess];
    for (int i = 0; i < nBlock; i++)
    {
        printf("Enter the size of block %i: ", i+1);
        scanf("%i", &sizeBlock);
        blockSize[i] = sizeBlock;
    }
    for (int i = 0; i < nProcess; i++)
    {
        printf("Enter the size of process %i: ", i+1);
        scanf("%i", &sizeProcess);
        processSize[i] = sizeProcess;
    }

    //allocation array
    for (int i = 0; i < nBlock; i++) {
            allocation[i] = -1;
    }
    for (int i = 0; i < nProcess; i++) {
            for (int j = 0; j < nBlock; j++) {
                if (blockSize[j] >= processSize[i]) { //if block size is greater than process size
                    allocation[i] = j; 
                    blockSize[j] -= processSize[i]; 
                    break;
                }
            }
        }
    
    printf("\nProcess No.\t\tProcess Size\t\t\tBlock no.\t\t\tTransffered Process Size\n");
    for (int i = 0; i < nProcess; i++)
    {
        printf("%i\t\t\t", i+1);
        printf("%i\t\t\t\t", processSize[i]);
        if (allocation[i] != -1)
        {
            printf("%i\t\t\t\t", allocation[i] + 1);
            printf("%i\t\t\t\t", processSize[i]);
        }
        else
        {
            printf("Not Allocated");
            printf("\t\t\t-");
        }
        printf("\n");
    }
    return 0;
}